Core/Rewriters/LastCallRewriter.cs

```diff
@@ -0,0 +1,170 @@
+﻿//  Copyright (c) rubicon IT GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using RhinoMocksToMoqRewriter.Core.Extensions;
+
+namespace RhinoMocksToMoqRewriter.Core.Rewriters
+{
+  public class LastCallRewriter : RewriterBase
+  {
+    public override SyntaxNode? VisitBlock (BlockSyntax node)
+    {
+      var trackedNodes = node.TrackNodes (node.DescendantNodes().OfType<StatementSyntax>(), CompilationId);
+      var baseCallNode = (BlockSyntax) base.VisitBlock (trackedNodes)!;
+
+      var rhinoMocksLastCallSymbol = Model.Compilation.GetTypeByMetadataName ("Rhino.Mocks.LastCall");
+      var rhinoMocksMockRepositorySymbol = Model.Compilation.GetTypeByMetadataName ("Rhino.Mocks.MockRepository");
+      if (rhinoMocksLastCallSymbol == null || rhinoMocksMockRepositorySymbol == null)
+      {
+        throw new InvalidOperationException ("Rhino.Mocks cannot be found.");
+      }
+
+      var lastCallExpressionStatementsInBlock = GetLastCallExpressionStatementsInBlock (baseCallNode, rhinoMocksLastCallSymbol).ToList();
+      if (lastCallExpressionStatementsInBlock.Count == 0)
+      {
+        return baseCallNode;
+      }
+
+      var nodesToBeRemoved = new List<SyntaxNode>();
+      foreach (var lastCallExpressionStatement in lastCallExpressionStatementsInBlock)
+      {
+        var currentNode = baseCallNode.GetCurrentNode (lastCallExpressionStatement.Node, CompilationId)!;
+        var lastCalledMock = GetLastCalledMockExpressionStatement (baseCallNode, lastCallExpressionStatement, rhinoMocksMockRepositorySymbol);
+        if (lastCalledMock == null)
+        {
+          Console.Error.WriteLine ($"WARNING: Unable to convert {lastCallExpressionStatement.Node}, last called mock was not inside the same method.");
```

Comment:
Please add the filepath and if possible line number to the error message. As for the output itself, we might change the stream later.